### PR TITLE
Add AnimatedProgress component

### LINKED ISSUE #979: Is pacman back in v6?
Some comments in the past said pacman is going to be back in v6. Did this happen?

I really love it.

### PR DESCRIPTION
**Pull request by sauliuskerusauskas**
### Description

- Added an `<AnimatedProgress />` component

https://user-images.githubusercontent.com/8139746/172902189-2c763576-7a96-4353-aa59-4ac517f2bd7a.mov

Has `color`, `size` and `pacmanColor` props. I didn't want to make assumptions so Pacman defaults to the `color` prop if not defined, but with `<Pacman pacmanColor="yellow" />` we can easily achieve:

https://user-images.githubusercontent.com/8139746/172902617-31f04464-2934-4b80-9a52-d70fc6e56b52.mov

The size of the circles is `7.5px` by default with a larger margin (to make space for the Pacman) vs `10px` for circles in the `Progress` component. This means the two can be swapped out easily at default sizing. One above the other for comparison:

compare

Also gracefully support presentations with a lot of slides that causes the circles to wrap:

https://user-images.githubusercontent.com/8139746/172903881-fd7f8ccc-800f-4c88-9183-aee2def760f7.mov

The way it works is by measuring where the active slides circle is in the DOM, relative to the it's internal container, then re-positions the Pacman over the top of it.

Completes #1105 and means we can close #979 and #926

#### Type of Change

Please delete options that are not relevant (including this descriptive text).

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### How Has This Been Tested?

Have tested across all modes manually. Have also copied existing and equally applicable Progress component tests, that cover the number of circles and the correct circle appearing as active, but testing the existence and the position of the Pacman is going to be difficult since it relies on real DOM measurements that aren't possible in JSDOM.

### CODE REVIEW

**Comment by carlos-kelly** [comment]
This is really great! Is it possible to make the regular progress dot disappear when the Pacman is over it? As in that regular progress dot is the current slide index.

**Comment by sauliuskerusauskas** [comment]
> This is really great! Is it possible to make the regular progress dot disappear when the Pacman is over it? As in that regular progress dot is the current slide index.

@carlos-kelly do you mean like this?

https://user-images.githubusercontent.com/8139746/173079148-67e82159-71c7-40a1-b903-e79a47a9aa16.mov

**Comment by sauliuskerusauskas** [comment]
> This is really great! Is it possible to make the regular progress dot disappear when the Pacman is over it? As in that regular progress dot is the current slide index.

@carlos-kelly do you mean like this?

https://user-images.githubusercontent.com/8139746/173079148-67e82159-71c7-40a1-b903-e79a47a9aa16.mov

**Comment by carlos-kelly** [comment]
@sauliuskerusauskas Yes! That's great.

File: src/components/animated-progress.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        setPacmanOffsetLeft(offsetLeft + halfOfCircleOccupiedSpace);
+        setPacmanOffsetTop(offsetTop + halfOfCircleOccupiedSpace);
+        setAlternateAnimation((alternateAnimation) => !alternateAnimation);
+      } else {
+        setPacmanOffsetLeft(null);
+        setPacmanOffsetTop(null);
+      }
+    }, [circleSize, activeCircleNode]);
+
+    const circleMargin = circleSize / 1.64;
```
Comment (carlos-kelly) [inline src/components/animated-progress.tsx:126]: I'd love to just hoist or note what this magic number is.

Comment (sauliuskerusauskas) [reply src/components/animated-progress.tsx]: 👍  Of course. Have improved it by adding the calculation in favour of the magic number here: https://github.com/FormidableLabs/spectacle/pull/1124/commits/5db35bd40b7d51148f607b5ba0740d2911ddb01a

**Pull request merged**